### PR TITLE
Fix #271: [trader bot] Trader approach1, since baseline trader bot doesn't actually trade yet

### LINKED ISSUE #271: [Trader bot] Add approach1 to do simplistic trading
### Status quo

The trader bot is implemented in [predictoor/trader/trader_agent.py](https://github.com/oceanprotocol/pdr-backend/blob/main/pdr_backend/trader/trader_agent.py). 

It has a `do_trade()` function. 
- It inputs two things:
  - the [Feed](https://github.com/oceanprotocol/pdr-backend/blob/main/pdr_backend/models/feed.py) object. Eg Binance ETH/USDT 5min
  - and the prediction: (amount staked up, amt staked down)
- With those inputs, it trades according to its implemented strategy 

Here's what the baseline `do_trade()` currently is:

```python
 async def do_trade(feed: Feed, prediction: Tuple[float, float]):
    """
    @description
        This function is called each time there's a new prediction available.
        By default, it prints the signal.
        The user should implement their trading algorithm here.
    @params
        feed : Feed
            An instance of the Feed object.

        prediction : Tuple[float, float]
            A tuple containing two float values, the unit is ETH:
            - prediction[0]: Amount staked for the price going up.
            - prediction[1]: Total stake amount.
    @note
        The probability of the price going up is determined by dividing
        prediction[0] by prediction[1]. The magnitude of stake amounts indicates
        the confidence of the prediction. Ensure stake amounts
        are sufficiently large to be considered meaningful.
    """
    pred_nom, pred_denom = prediction
    print(f"      {feed} has a new prediction: {pred_nom} / {pred_denom}.")
    # Trade here
    # ...
```

### The problem

Look at the bottom of `do_trade()`. It doesn't actually _do_ any trading yet. Not even the simplest possible trading strategy: "buy if prediction is up; exit 5 min later".

We expect >1 trading strategies over time.

Predictoor bots handle this by having "approach1", "approach2", etc. We could do the same for trader bots.

### DoD

- [x] Implement predictoor/trader/approach1/, where it implements `do_trade()` with the trading strategy: "buy if prediction is up; exit 5 min later". 
- [x] Have good unit tests
- [x] Test that it works for real on mainnet

### Related issues

- [pdr-backend#231](https://github.com/oceanprotocol/pdr-backend/issues/231) used to have "add strategy" plus "do backtesting". But 271 (_this_ issue) does that. So 231 was de-scoped to just "do backtesting"

### PR DESCRIPTION
**Pull request by trentmc**
Fixes #271

Note: Roberto had created the branch. I simply created the PR so that I could give comments on the code. It's still Roberto's work:)

### CODE REVIEW

File: pdr_backend/trader/approach1/trader_agent1.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
+import random
```
Comment (trentmc) [inline pdr_backend/trader/approach1/trader_agent1.py:1]: I like that you're implementing this by inheriting from the base agent, and creating approach1. Perfect.

That's in line with what predictoor does, and what [pdr-backend#231](https://github.com/oceanprotocol/pdr-backend/issues/231) anticipates: "[trader bot] Streamline strategy implementation, introduce backtesting". Doing it here first (for issue 275) is a nice precursor to 231. Great!

Comment (idiom-bytes) [reply pdr_backend/trader/approach1/trader_agent1.py:1]: Thank you for taking this from the depths of branches and deploying a draft, and providing feedback. Cheers!

File: scripts/pm2-config/localnet-predictoor.config.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
```
[the remark this answers is not in the document]
Comment (idiom-bytes) [reply scripts/pm2-config/localnet-predictoor.config.js:1]: 100%

**Comment by idiom-bytes** [comment]
[MEXC API + Predictoor Feed Challenges]
- Most USDT pairs (BTC, ETH, XRP) are not available to trade via their API/[UI-Allow-List].
- Their [UI-Allow-List] offers BTC/USDC, however this market does not get listed in the CCXT library
- To verify the CCXT implementation is working, I need a test/real market to validate. Searching...
- MEXC python lib may offer perps/USDC :shrug: however I'm hoping to wrap this with a CCXT implementation, so it should hopefully work with BNB.

**Comment by idiom-bytes** [comment]
-> Bot is trading w/ MEXC.
-> Will run black/pylint and get the tests passing.
-> Additional work has been branched into Agent2, and pushed as a draft PR.
image

**Comment by idiom-bytes** [comment]
Let's see what happens...
image

**Review by idiom-bytes — APPROVED** [review]
LGTM. 
@trentmc, please review

File: pdr_backend/trader/approach1/trader_agent1.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            # close it
+            order = self.exchange.create_market_sell_order(
+                self.config.exchange_pair, amount
+            )
+
+            print(f"     [Trade Closed] {self.exchange}")
+            print(f"     [Previous Order] {self.order}")
+            print(f"     [Closing Order] {order}")
+
+            # TODO - Calculate PNL (self.order - order)
```
[the remark this answers is not in the document]
Comment (idiom-bytes) [reply pdr_backend/trader/approach1/trader_agent1.py:102]: No, I'll remove it.

File: pdr_backend/trader/approach1/trader_agent1.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+            # If order is successful, we log the order so we can close it
+            if order != None and isinstance(order, dict):
+                self.order = order
+                print(f"     [Trade Opened] {self.exchange}")
+                print(f"     [Opening Order] {order}")
+        else:
+            print(
+                f"     [No Trade] prediction does not meet requirements: {pred_properties}"
+            )
```
[the remark this answers is not in the document]
Comment (idiom-bytes) [reply pdr_backend/trader/approach1/trader_agent1.py:125]: I specifically didn't want to add any logic around the trading strategy to improve it. 

"Leaving it up to the reader"

Hopefully it's ok if I omit this.

**Comment by idiom-bytes** [comment]
I have now:
1. Created a different file for test_ccxt_exchanges
2. Created new test coverage for test_trader_agent1.py, which includes async tests which were not passing
3. Fixed existing test coverage inside test_trader_agent.py that was being skipped by CICD, such that it's now executed correctly and passing
4. Updated setup.y so `pytest-asyncio` is now being installed

**Review by idiom-bytes — APPROVED** [review]
LGTM

**Pull request merged**